### PR TITLE
[Locked Figures Labels] Make labels optional to increase type safety

### PR DESCRIPTION
**Pull request by nishasy**
## Summary:
Since the `labels` field was added to the interactive graph
locked figures after they were released and used by content
authors, there are already exercises saved with locked figures
that don't have a `labels` field.

If we make this a required field, this can lead to issues with
the existing exercises in the future. To increase type safety,
I'm making `labels` optional. We should also do this with
new types added in the future. (Labels haven't been released
for use yet, so we can still fix this now without issue.)

Issue: none

## Test plan:
`yarn jest`
`yarn typecheck`

### CODE REVIEW

File: packages/perseus/src/widgets/interactive-graphs/graph-locked-labels-layer.tsx [hunk 1]
```diff
@@ -37,22 +37,22 @@ export default function GraphLockedLabelsLayer(props: Props) {
         ) {
             return (
                 <React.Fragment key={i}>
-                    {figure.labels.map((label, j) => (
+                    {(figure.labels ?? []).map((label, j) => (
```
Comment (jeremywiebe) [inline packages/perseus/src/widgets/interactive-graphs/graph-locked-labels-layer.tsx:40]: This works, but could we just do... 
```suggestion
                    {figure.labels?.map((label, j) => (
```

File: packages/perseus/src/widgets/interactive-graphs/interactive-graph-question-builder.ts [hunk 2]
```diff
@@ -324,13 +324,15 @@ class InteractiveGraphQuestionBuilder {
...
-            })),
+            labels: options?.labels
+                ? options.labels.map((label) => ({
+                      type: "label",
+                      coord: label.coord ?? vec.midpoint(point1, point2),
+                      text: label.text,
+                      color: options?.color ?? "grayH",
+                      size: label.size ?? "medium",
+                  }))
+                : undefined,
```
Comment (jeremywiebe) [inline packages/perseus/src/widgets/interactive-graphs/interactive-graph-question-builder.ts:335]: We can use optional chaining here again for the same result: 

```suggestion
            labels: options?.labels?.map((label) => ({
                      type: "label",
                      coord: label.coord ?? vec.midpoint(point1, point2),
                      text: label.text,
                      color: options?.color ?? "grayH",
                      size: label.size ?? "medium",
                  })),
```

File: packages/perseus/src/widgets/interactive-graphs/interactive-graph-question-builder.ts [hunk 3]
```diff
@@ -362,13 +364,15 @@ class InteractiveGraphQuestionBuilder {
...
             color: options?.color ?? "grayH",
             points: [tail, tip],
-            labels: (options?.labels ?? []).map((label) => ({
-                type: "label",
-                coord: label.coord ?? vec.midpoint(tail, tip),
-                text: label.text,
-                color: options?.color ?? "grayH",
-                size: label.size ?? "medium",
-            })),
+            labels: options?.labels
```
Comment (jeremywiebe) [inline packages/perseus/src/widgets/interactive-graphs/interactive-graph-question-builder.ts:367]: Same note about optional chaining...

**Review by jeremywiebe — APPROVED** [review]
I think you can tighten up quite a few lines in the question builder using optional chaining.  Otherwise, this looks great!

File: packages/perseus/src/widgets/interactive-graphs/graph-locked-labels-layer.tsx (hunk 1 above)
Comment (nishasy) [reply packages/perseus/src/widgets/interactive-graphs/graph-locked-labels-layer.tsx:40]: Huh... for some reason I was convinced that you couldn't use `map` on something that could be undefined 🤔 Good to know.

**Pull request merged**